src/ninshiki_cpp/detector/detector.cpp

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2021 ICHIRO ITS
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "ninshiki_cpp/detector/detector.hpp"
+
+namespace ninshiki_cpp::detector
+{
+
+Detector::Detector(bool gpu, bool myriad)
+{
+  file_name = static_cast<std::string>(getenv("HOME")) + "/yolo_model/obj.names";
+  std::string config = static_cast<std::string>(getenv("HOME")) + "/yolo_model/config.cfg";
+  std::string weights = static_cast<std::string>(getenv("HOME")) +
+    "/yolo_model/yolo_weights.weights";
+  net = cv::dnn::readNet(weights, config, "");
```

Comment:
just a minor thing, maybe change this into `model` instead of `weights `

Reply:
ok